src/Array.jl

```diff
@@ -132,6 +132,8 @@ function Matrix{T}(g::Gate{<:SU{N}}) where {T, N}
     return g.array |> Matrix{T}
 end
 
+Array{T}(g::Gate{SU{N}}) where {T,N} = Array{T,2 * length(g)}(reshape(Matrix{T}(g), fill(2, 2 * Int(log2(N)))...))
+
```

Comment:
Please move this method down below, don't put it first.
The method order should be more or less similar to the order of definition of `Gate`s.